Semantic definitions for alignment (business terminology)

apinzenoehler opened the pull request:
Horizontal segment formulas complete,
some documentation in horizontal pending
Horizontal segment constants and Cant segment constants disentangled
Cant overview done, Cant segment formulas pending

apinzenoehler committed:
Semantic definitions for alignment (business terminology)

Horizontal segment formulas complete,
some documentation in horizontal pending
Horizontal segment constants and Cant segment constants disentangled
Cant overview done, Cant segment formulas pending

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+
+1. Contemporary engineering usually establishes first a horizontal layout in a properly projected plane.
+
+2. In a second step the vertical profile (i.e. sequence of segments with constant gradients) is added.
```
Is it only a sequence of constant gradients? When are the ones with changing gradients added?

apinzenoehler replied:
Well spotted. Will modify.

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+
+3. In the rail domain in most cases a cant layout is added to the horizontal layout to compensate a part of the unwanted lateral acceleration.
+
+4. In a final step the proposed layout is checked against a defined set rules, formulas and threshold to guarantee the conformance against the regulation.
```
```suggestion
4. In a final step the proposed layout is checked against a defined set of rules, formulas and thresholds to guarantee the conformance against the regulation.
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+
+4. In a final step the proposed layout is checked against a defined set rules, formulas and threshold to guarantee the conformance against the regulation.
+
+The sequence of the steps might change from case to case and might be repeated one or more times to achieve the economic objectives and fullfill regulatory safety requirements. 
```
```suggestion
The sequence of steps might change from case to case and might be repeated one or more times to achieve the economic objectives and fulfill regulatory safety requirements. 
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+The sequence of the steps might change from case to case and might be repeated one or more times to achieve the economic objectives and fullfill regulatory safety requirements. 
+
+
+Contemporary alignment design itself implements almost always a 2.5 dimension approach. 
```
```suggestion
Contemporary alignment design almost always implements a 2.5 dimension approach. 
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+
+Contemporary alignment design itself implements almost always a 2.5 dimension approach. 
+
+The resulting and documented geometry might be very precise or just good enough to meet safety thresholds. This depends on factors like priorities of the management, date of the design - existing alignments might have been more then 50 years ago - or software tools used. Working with legacy data in a high precision BIM model requires a good understanding of these factors.
```
```suggestion
The resulting and documented geometry might be very precise or just good enough to meet safety thresholds. This depends on factors like priorities of the management, date of the design - existing alignments might have been designed more then 50 years ago - or software tools used. Working with legacy data in a high precision BIM model requires a good understanding of these factors.
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+According to IFC modeling principles alignment entities are organised in two large parts. The two parts work together, but they can also be used independently from each other
+
+1. Business aspects of alignment
+2. Representation with IFC geometry
```
```suggestion
2. Representation with the IFC geometry resources
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+1. Business aspects of alignment
+2. Representation with IFC geometry
+
+**Business aspects of alignment:** Here the focus is a on a schema structure as close to business terminology as possible. Here it is possible to have a very detailed segment structure with many domain specific properties attached. Examples for domain specific properties are design speed or cant deficiency.
```
```suggestion
**Business aspects of alignment:** Here the focus is a on a schema structure as close to business terminology as possible. It is possible to have a very detailed segment structure with many domain specific properties attached. Examples for domain specific properties are design speed or cant deficiency.
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+
+**Business aspects of alignment:** Here the focus is a on a schema structure as close to business terminology as possible. Here it is possible to have a very detailed segment structure with many domain specific properties attached. Examples for domain specific properties are design speed or cant deficiency.
+
+**Representation with IFC geometry:** Here the focus is on using as much of the established IFC geometry entities as possible. A mapping between Business aspects and IFC geometry is proposed.
```
```suggestion
**Representation with IFC geometry resources:** Here the focus is on using as much of the established IFC geometry entities as possible. A mapping between Business aspects and IFC geometry is proposed.
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
-A single alignment may have:  
+In IFC a single alignment must have:  
   
 \* a horizontal alignment defined in the x/y plane of the engineering coordinate system  
```
```suggestion
* a horizontal alignment defined in the x/y plane of the engineering coordinate system  
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+
+A single alignment may have: 
+
 \* an accompanying vertical alignment, defined along the horizontal alignment in the distance along / z coordinate space  
```
```suggestion
* an accompanying vertical alignment, defined along the horizontal alignment in the distance along / z coordinate space  
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+A single alignment may have: 
+
 \* an accompanying vertical alignment, defined along the horizontal alignment in the distance along / z coordinate space  
 \* a relative alignment, defined as a span within another alignment and/or at constant or variable offsets  
```
```suggestion
* a relative alignment, defined as a span within another alignment and/or at constant or variable offsets  
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+
 \* an accompanying vertical alignment, defined along the horizontal alignment in the distance along / z coordinate space  
 \* a relative alignment, defined as a span within another alignment and/or at constant or variable offsets  
 \* a 3D alignment, either computed from the horizontal and vertical alignment, or extracted from geospatial data.  
```
```suggestion
* a 3D alignment, either computed from the horizontal and vertical alignment, or extracted from geospatial data.  
```

SergejMuhic committed:
Correcting md in alignment

SergejMuhic commented on IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md:
```diff
@@ -0,0 +1,59 @@
+
+
+
+The geometric perspective denotes in the context of the business terminology related IfcAlignment documentation the traditional view. Before the availability of modern computers alignment design was performed using "traditional" drawing techniques. In the first phase of computerization this origin led to a representation in the x,y space first and a check of safety related properties in a second  ...
```
Had to switch this up. 😄 
```suggestion
The traditional view is denoted by the geometric perspective in the context of the business terminology related IfcAlignment documentation. Before the availability of modern computers alignment design was performed using "traditional" drawing techniques. In the first phase of computerization this origin led to a representation in the x,y space first and a check of safety related properties in a second step. This can still be seen in regulations which have been put into effect 1980 or earlier. Of course designs which have been produced on basis of these regulations reflect the "good enough" attitude in the precision of the documentation. 
```

SergejMuhic commented on IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md:
```diff
@@ -0,0 +1,59 @@
+
+**Generic calculation of intrinsic x, y-coordinates for a given curvature**
+
+For each horizontal alignment segment with known curvature formula a generic method to calculate segment intrinsic coordinates exists.
```
```suggestion
For each horizontal alignment segment with a known curvature formula a generic method to calculate segment intrinsic coordinates exists.
```

SergejMuhic commented on IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md:
```diff
@@ -0,0 +1,59 @@
+
+!["Double integration"](../../figures/ifcalignmenthorizontalsegmenttypeenum-curvature2coord.png "Figure 1 &mdash; Double integration of curvature yielding intrinsic coordinates")
+
+NOTE:&nbsp;While it is possible to apply the generic calculation also for trivial cases like LINE or CIRCULARARC it is much more efficient to use available formulas.
```
```suggestion
>NOTE:&nbsp;While it is possible to apply the generic calculation also for trivial cases like LINE or CIRCULARARC it is much more efficient to use available formulas.
```

SergejMuhic commented on IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md:
```diff
@@ -0,0 +1,59 @@
+
+**Word or warning**
+
+"Good enough" traditional designs have to be carefully checked before being included into a high precision 3D model. Intermediate corrections might be necessary. Fortunately the clothoid works very well with comparable documentation quality both in the classic geometric perspective and in the more recent kinematic perspective. Fortunately the huge majority of horizontal transition bends are desig ...
```
```suggestion
"Good enough" traditional designs have to be carefully checked before being included into a high precision 3D model. Intermediate corrections might be necessary. Fortunately the clothoid works very well with comparable documentation quality both in the classic geometric perspective and in the more recent kinematic perspective. Fortunately the vast majority of horizontal transition bends are designed and implemented as clothoids.
```

SergejMuhic commented on IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md:
```diff
@@ -0,0 +1,59 @@
+
+NOTE:&nbsp;While it is possible to apply the generic calculation also for trivial cases like LINE or CIRCULARARC it is much more efficient to use available formulas.
+
+**Word or warning**
```
```suggestion
**Word of warning**
```

SergejMuhic commented on IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md:
```diff
@@ -0,0 +1,59 @@
+
+**Recommendation**
+
+Check the relevant regulations for the network in question. Alignment designs as such are very stable over the livetime of the road or track. Especially for old designs quality and precision of available documentation has to be checked very carefully. A clear understanding of limitations should be established before implementing automated data flows between high precision BIM environments and leg ...
```
```suggestion
Check the relevant regulations for the network in question. Alignment designs as such are very stable over the lifetime of the road or track. Especially for old designs quality and precision of available documentation has to be checked very carefully. A clear understanding of limitations should be established before implementing automated data flows between high precision BIM environments and legacy documentation systems. This applies both to legacy, central databases and to legacy, individual documents. 
```

SergejMuhic commented on IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md:
```diff
@@ -0,0 +1,59 @@
+
+**Intrinsic coordinate, intrinisc coordinate system of an alignment segment:**
+
+The origin of an intrinsic coordinate system is the start of the segment. The direction of the positive x-axis is the startdirection of the segments.
```
```suggestion
The origin of an intrinsic coordinate system is the start of the segment. The direction of the positive x-axis is the start direction of the segments.
```

SergejMuhic reviewed (changes requested):
Really good work. Some remarks_
1. Some grammatical errors and typos, proposals are available for acceptantce
2. Committed part of the corrected IfcAlignment documentation directly to your branch, if you could add the commit to this PR would be wonderful
3. Need some input from @larswik 
4. I would use the curvature2coord.png in the IfcSpiral documentation.

apinzenoehler committed:
Update IFC4x3/Constants/b/BLOSSCURVE_2PSkBIAo5DrBxM4nxjUkwb.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler commented on IFC4x3/Constants/b/BLOSSCURVE_0hYbIcTB5DJhg8TxL3hntC.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="BLOSSCURVE_0hYbIcTB5DJhg8TxL3hntC" Name="BLOSSCURVE" UniqueId="2b8a54a6-74b1-4d4e-ba88-77b543af1dcc" />
```
Yes this is a correct observation and it was done on purpose. See commit comment "Cant segment constants disentangled". It was caused by the current Enum constants naming in RC4 for IfcAlignmentHorizontalSegmentTypeEnum and IfcAlignmentCantSegmentTypeEnum. Having only on (IfcDoc) constant for BLOSSCURVE is a little bit unfortunate for proper documentation of high performance transition bends.

In EN 13803 it is stated that for Linear (aka Clothoid) transition, Helmert, Bloss, Cosine and Sine the base formula for curvature variations and the base formula for cant transitions are structurally identical. Viennese bend is the one example where formula for horizontal and the formula for cant transitions are structurally different.

BUT: although the formulas are structurally the same, different symbols are in use. Small greek "kappa" for curvature and capital D for cant. So the formulas look different. There is a commit due adding the cant formulas this evening.

Fortunately in RC4 we already use different names for the linear transition, namely CLOTHOID in horizontal layout and LINEARTRANSITION in cant layout, so it is not an issue there.

I see two options:

1) keep two seperate, nevertheless identically named constants, one for the horizontal documentation and one for the cant documentation.

2) rename the constants for cant. E.g. BLOSSCURVECANT instead of BLOSSCURVE .

I need guidance here.

PS: please note that a similar issue exists for CLOTHOID in horizontal and vertical layouts. The documentation for CLOTHOID in vertical will be different from the documentation for CLOTHOID in horizontal.

SergejMuhic replied:
Ahh, sorry. Forgot about that. No need for removing then.

apinzenoehler committed:
Update IFC4x3/Constants/b/BLOSSCURVE_2PSkBIAo5DrBxM4nxjUkwb.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Constants/c/CIRCULARARC_38N4a1WJ509ud6zzClEh4g.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Constants/c/CLOTHOID_0MF3ueksjFphb$uHgZP5g6.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Constants/c/CLOTHOID_0MF3ueksjFphb$uHgZP5g6.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Constants/c/CLOTHOID_0MF3ueksjFphb$uHgZP5g6.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler commented on IFC4x3/Constants/c/COSINECURVE_1EPwu89RnE2wA$LkEQYGUK.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="COSINECURVE_1EPwu89RnE2wALkEQYGUK" Name="COSINECURVE" UniqueId="4e67ae08-25bc-4e0b-a2bf-56e39a890794" />
```
Same answer as for the bloss ;-)

apinzenoehler commented on IFC4x3/Constants/c/CUBICSPIRAL_21hOPMUMn1mvG3F$Y7iNv3.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CUBICSPIRAL_21hOPMUMn1mvG3FY7iNv3" Name="CUBICSPIRAL" UniqueId="81ad8656-796c-41c3-9403-3ff887b17e43">
-	<Documentation>Cubic spiral transition.</Documentation>
+	<Documentation>Cubic spiral transition. Enter description of &quot;Transition curves in Road Design&quot;</Documentation>
```
According to decision in last Friday's technical session the CUBICSPIRAL will be removed.

larswik replied:
I think that we should maybe ask @jmirtsch? We have only used clothoids in Sweden.

apinzenoehler commented on IFC4x3/Constants/c/CUBIC_2zPyKsqGvAMAYozqlaSFKO.xml:
```diff
@@ -1,5 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CUBIC_2zPyKsqGvAMAYozqlaSFKO" Name="CUBIC" UniqueId="bd67c536-d10e-4a58-a8b2-f74be470f518">
-	<Documentation>y = a x³</Documentation>
+	<Documentation>y = a x³
```
Good question. 

Anyhow documentation for CUBIC needs to be extended.

Here a comment/review by @jmirtsch and Lee Gregory would be useful.

larswik replied:
Why not use the full polynomial and b=c=d=0 if needed?

apinzenoehler committed:
Update IFC4x3/Constants/h/HELMERTCURVE_14V3RUc6LFh9bZo1rEFhxL.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Constants/h/HELMERTCURVE_14V3RUc6LFh9bZo1rEFhxL.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler commented on IFC4x3/Constants/h/HELMERTCURVE_1qak5zKIL9hepxekxNx$60.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HELMERTCURVE_1qak5zKIL9hepxekxNx60" Name="HELMERTCURVE" UniqueId="7492e17d-5125-49ae-8cfb-a2eed7eff180" />
```
Same answer as for the bloss ;-)

apinzenoehler committed:
Update IFC4x3/Constants/s/SINECURVE_33lTdYQrv7DB7H9wXvDAvN.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler commented on IFC4x3/Constants/s/SINECURVE_3jBKjsnmX4QA73Bh_qTnj0.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SINECURVE_3jBKjsnmX4QA73Bh_qTnj0" Name="SINECURVE" UniqueId="ed2d4b76-c708-4468-a1c3-2ebfb4771b40">
```
Same answer as for the bloss ;-)

apinzenoehler commented on IFC4x3/Constants/v/VIENNESEBEND_2Z8G0iUzb0sQDw_7$tEcwv.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VIENNESEBEND_2Z8G0iUzb0sQDw_7tEcwv" Name="VIENNESEBEND" UniqueId="a321002c-7bd9-40d9-a37a-f87ff73a6eb9">
```
Same answer as for the bloss ;-)

apinzenoehler commented on IFC4x3/Constants/v/VIENNESEBEND_1wHLHEhpb2GhA0VAasbHUK.xml:
```diff
@@ -1,3 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VIENNESEBEND_1wHLHEhpb2GhA0VAasbHUK" Name="VIENNESEBEND" UniqueId="7a45544e-af39-4242-b280-7ca936951794" />
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VIENNESEBEND_1wHLHEhpb2GhA0VAasbHUK" Name="VIENNESEBEND" UniqueId="7a45544e-af39-4242-b280-7ca936951794">
+	<Documentation>**Curvature formula**
```
Upcoming .... see note in commit message "some documentation in horizontal pending" :-)

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler committed:
Update IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentHorizontalSegmentTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

apinzenoehler commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,10 +1,63 @@
+For the purposes of IFC the English term "alignment" defines three essentially separate but closely interconnected concepts.
```
Thanks for this feedback.

apinzenoehler committed:
Feedback from PR-Review

Typo in constant for HELMERTCURVE (this kind of typos aborts loading in IfcDoc tool)
Addtional documentation for COSINECURVE, SINECURVE, VIENNESEBEND
Improvement in documentation IfcAlignment
Finalisation of CUBIC (+ removal of CUBICSPIRAL) is pending, will become a separate PR.

apinzenoehler committed:
Merge branch 'bSI-InfraRoom:develop' into alignment_semantic_documentation

larswik commented on IFC4x3/Constants/c/CIRCULARARC_38N4a1WJ509ud6zzClEh4g.xml:
```diff
@@ -1,5 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CIRCULARARC_38N4a1WJ509ud6zzClEh4g" Name="CIRCULARARC" UniqueId="c85c4901-8131-4027-89c6-f7d32f3ab12a">
-	<Documentation>Circular arc transiiton.</Documentation>
+	<Documentation>In the geometric perspective, it denotes a connection between two points that follows a circular path. In the dynamic perspective, it denotes a segment with constant lateral acceleration on the moving train or vehicle, i.e. constant curvature.
```
Maybe: "A continuous section of the circumference of a circle"?
Isn't a train a kind of vehicle?

SergejMuhic replied:
There has been a change suggested and accepted. Could you have a look at the new text?

On cubic, I agree. I would modify it.

larswik commented on IFC4x3/Constants/l/LINE_0aQELtHxv8tPRsAoCd1LjP.xml:
```diff
@@ -1,5 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="LINE_0aQELtHxv8tPRsAoCd1LjP" Name="LINE" UniqueId="2468e577-47be-48dd-96f6-2b2327055b59">
-	<Documentation>Straight connection between two points.</Documentation>
+	<Documentation>In the geometry perspective it denotes a straight connection between two points. In the dynamic perspective, it denotes a segment with a curvature with a value of 0. This means that no lateral acceleration acts on the moving train or vehicle.
```
Is a train not a vehicle?

apinzenoehler replied:
The term "train" might be a little bit vague here. 

Most commonly it is used to denote an assembly of vehicles which from rail operations view (e.g. planning, dispatching, securing safe movement, accounting) act as a (temporary) unit. In common rail the vehicles are categorized into locomotives and wagons, the locomotives sometimes also carrying passengers or freight (railcar).

@SergejMuhic : do we need definitions of this kind? If yes, should it be placed near the usage of the term in type and entitiy definitions or should this go into chapter 3 "Terms, definitions, and abbreviated terms"?

apinzenoehler replied:
Concerning Cubic: in the literature it is almost always discussed with the 3rd order term only. The main problem here is that y is a function of x,  whereas in the other transition types, both x and y are a function of s.

I think that indeed it is OK to state the full polynomial in the definition. The documentation should point out the 3rd order term only aspect and also explain the y=f(x) versus y=f(s) aspect. Here one could also mention engineering shortcuts like approximating x with s, known in some parts of the world as the cubic spiral.

@SergejMuhic OK?

SergejMuhic replied:
Regarding train/vehicle, the shared definition says it already 

> Most commonly it is used to denote an assembly of vehicles

so, I would remove train and just leave vehicle as it is a hypernym of train.

apinzenoehler replied:
"Train" will be removed,

Concerning CUBICSPIRAL I hope to get a final advice from a top track engineering expert today. --> I propose to isolate removal of CUBICSPIRAL and documentation into a separate small PR.

larswik commented on IFC4x3/Constants/v/VIENNESEBEND_1wHLHEhpb2GhA0VAasbHUK.xml:
```diff
@@ -1,3 +1,10 @@
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VIENNESEBEND_1wHLHEhpb2GhA0VAasbHUK" Name="VIENNESEBEND" UniqueId="7a45544e-af39-4242-b280-7ca936951794">
+	<Documentation>The Viennese Bend (R) is an innovative track geometry transition element. Instead of analyzing the vehicle movement at the track plane the optimization efforts target a gravity center line at a defined height above the rails.
+
+As a consequence the path of the horizontal alignment center line is also influenced by the cant layout. The first part of the curvature formula is assembled from the basic function like the other transition bends. The additional term contains the bank angle &quot;&amp;psi;&quot; and the gravity center line height &quot;h&quot; and is unique to the Viennes Bend (R). This term causes a small movem ...
```
height=>elevation?

apinzenoehler replied:
Hmmm ... in this case the original German term is "Trassierungshöhe". It denotes a local geometry property placing the line central to kinematic analysis relative to track plane. I am not a native speaker but discussion  in https://en.wikipedia.org/wiki/Height makes me think that both "elevation" and "altitude" have a slightly different meaning.

SergejMuhic replied:
I think in this context 'height' is the correct term.

larswik commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,25 +1,78 @@
+* a relative alignment, defined as a span within another alignment and/or at constant or variable offsets  
+* a 3D alignment, either computed from the horizontal and vertical alignment, or extracted from geospatial data.  
   
 Alignments may be aggregated into referents (_IfcReferent_) or derivative alignments. Derivative alignments may be used to indicate dependent alignments, such as an alignment for a bridge that is relative to a parent alignment for a road, where the child _IfcAlignment_ may have _Axis_ set to _IfcOffsetCurveByDistances_ that starts and ends at a span within the extent of the original _IfcAlignment ...
```
What does _Axis_ refer to here?

apinzenoehler replied:
This is previous text. Please provide a better description based on the final schema. @larswik @czapplitec ?

SergejMuhic replied:
New issue already created from one of the comments below.

larswik commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,25 +1,78 @@
   
 Alignments may be assigned to groups using _IfcRelAssignsToGroup_, where _IfcGroup_ or subtypes may capture information common to multiple alignments.  
   
 Supported representations of <span class="self-ref">IfcAlignment</span>.Axis are:  
```
_Axis_ again. Should this instead be "Supported shape representations for IfcAlignment are (RepresentationIdentifier='Axis" and RepresentationType="Curve2D/Curve3D"):
...

apinzenoehler replied:
This is previous text. Please provide a better description based on the final schema. @larswik @czapplitec ?

SergejMuhic replied:
We can address this after merging this PR. It should be a smaller change.

larswik commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md:
```diff
@@ -1,25 +1,78 @@
+* _IfcPolyline_ as a 3D alignment by a 3D polyline representation (such as coming from a survey)  
+* _IfcPolyline_ as a 2D horizontal alignment by a 2D polyline representation (such as in very early planning phases or as a map representation)  
   
 > NOTE&nbsp; Although _Axis_ is an _IfcCurve_ base type, only derived types _IfcGradientCurve_, _IfcSegmentedReferenceCurve_, _IfcOffsetCurveByDistances_, and _IfcPolyline_ are meant to be supported types. Derivative specifications (Model View Definitions) may expand this set to include additional curve types.
```
_Axis_

apinzenoehler replied:
This is previous text. Please provide a better description based on the final schema. @larswik @czapplitec ?

apinzenoehler committed:
Minor documentation change in constant LINE

"train" removed

larswik commented:
I understood that once this PR has been merged, I will create a new PR addressing #138

apinzenoehler committed:
Removed "train" from documentation CIRCULARARC
